common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java

```diff
@@ -0,0 +1,77 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sedona.common.raster;
+
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.junit.Test;
+import org.opengis.referencing.FactoryException;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class RasterBandAccessorsTest extends RasterTestBase {
+
+    @Test
+    public void testBandNoDataValueCustomBand() throws FactoryException {
+        int width = 5, height = 10;
+        GridCoverage2D emptyRaster = RasterConstructors.makeEmptyRaster(1, width, height, 53, 51, 1, 1, 0, 0, 4326);
+        double[] values = new double[width * height];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = i + 1;
+        }
+        emptyRaster = MapAlgebra.addBandFromArray(emptyRaster, values, 2, 1d);
+        assertNotNull(RasterBandAccessors.getBandNoDataValue(emptyRaster, 2));
+        assertEquals(1, RasterBandAccessors.getBandNoDataValue(emptyRaster, 2), 1e-9);
+        assertNull(RasterBandAccessors.getBandNoDataValue(emptyRaster));
+    }
+
+    @Test
+    public void testBandNoDataValueDefaultBand() throws FactoryException {
+        int width = 5, height = 10;
+        GridCoverage2D emptyRaster = RasterConstructors.makeEmptyRaster(1, width, height, 53, 51, 1, 1, 0, 0, 4326);
+        double[] values = new double[width * height];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = i + 1;
+        }
+        emptyRaster = MapAlgebra.addBandFromArray(emptyRaster, values, 1, 1d);
+        assertNotNull(RasterBandAccessors.getBandNoDataValue(emptyRaster));
+        assertEquals(1, RasterBandAccessors.getBandNoDataValue(emptyRaster), 1e-9);
+    }
+
+    @Test
+    public void testBandNoDataValueDefaultNoData() throws FactoryException {
+        int width = 5, height = 10;
+        GridCoverage2D emptyRaster = RasterConstructors.makeEmptyRaster(1, width, height, 53, 51, 1, 1, 0, 0, 4326);
+        double[] values = new double[width * height];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = i + 1;
+        }
+        assertNull(RasterBandAccessors.getBandNoDataValue(emptyRaster, 1));
+    }
+
+    @Test
+    public void testBandNoDataValueIllegalBand() throws FactoryException, IOException {
+        GridCoverage2D raster = rasterFromGeoTiff(resourceFolder + "raster/raster_with_no_data/test5.tiff");
```

Comment:
`test5` has a file extension `TIFF` but you specified `tiff` here. Please change the file extension of the original image to `tiff`.

Reply:
That didn't seem to matter for my local tests. However, I've renamed the file extension to tiff and pushed now